editor/sidebar/featured-image/index.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { __ } from 'i18n';
+import Button from 'components/button';
+import PanelBody from 'components/panel/body';
+import MediaUploadButton from 'blocks/media-upload-button';
+import Spinner from 'components/spinner';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class FeaturedImage extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			media: null,
+			loading: false,
+		};
+
+		// This boolean should be removed when we provide QueryComponents
+		this.mounted = true;
+	}
+
+	componentDidMount() {
+		this.fetchMedia();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.featuredImageId !== this.props.featuredImageId ) {
+			this.fetchMedia();
+		}
+	}
+
+	componentWillUnmount() {
+		this.mounted = false;
+	}
+
+	fetchMedia() {
+		if ( ! this.props.featuredImageId ) {
+			this.setState( {
+				media: null,
+				loading: false,
+			} );
+			return;
+		}
+		this.setState( {
+			media: null,
+			loading: true,
+		} );
+		const mediaIdToLoad = this.props.featuredImageId;
+		new wp.api.models.Media( { id: mediaIdToLoad } ).fetch()
```

Comment:
`Backbone.Model#fetch` returns a `jqXHR` ([reference](http://backbonejs.org/#Model-fetch)), which itself is a superset of `XMLHttpRequest` ([reference](http://api.jquery.com/jQuery.ajax/#jqXHR)), which makes available an `abort` method ([reference](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort)).

Maybe rather than tracking `this.mounted`, we can store a reference to the request here and abort it if the component unmounts? I'm not sure what affect this has on promise resolution though.